src/snovault/elasticsearch/simple_queue.py

```diff
@@ -0,0 +1,156 @@
+'''
+Simple uuid queue for indexing process
+- Can be overridden by another script, class, module as long as it exposes
+the proper functions used in the indexer classes.
+'''
+import time
+from os import getpid
+
+
+class SimpleUuidServer(object):
+    '''Simple uuid queue as a list'''
+    def __init__(self):
+        self._uuids = []
+        self._client_conns = {}
+        self.errors = []
+
+    def _get_uuid(self):
+        if self._uuids:
+            return self._uuids.pop()
+        return None
+
+    def _load_uuid(self, uuid):
+        '''
+        Load uuid into queue
+        - Uuid ordering impacts indexing time, list is reversed due to
+        pop being used in get uuid
+        '''
+        self._uuids.insert(0, uuid)
+
+    def get_uuids(self, client_id, cnt):
+        '''The only way to get uuids from queue'''
+        uuids = []
+        if client_id not in self._client_conns:
+            self._client_conns[client_id] = {
+                'uuid_cnt': 0,
+                'get_cnt': 0,
+                'results': None,
+            }
+        client_conn = self._client_conns[client_id]
+        client_conn['get_cnt'] += 1
+        if cnt == -1:
+            cnt = len(self._uuids)
+        if cnt and not client_conn['uuid_cnt']:
+            # There are uuids on the server and the worker is not working
+            while cnt > 0:
+                uuid = self._get_uuid()
+                if not uuid:
+                    break
+                uuids.append(uuid)
+                cnt -= 1
+            client_conn['uuid_cnt'] = len(uuids)
+        return uuids
+
+    def has_uuids(self):
+        '''Are there uuids in the queue'''
+        if self._uuids:
+            return True
+        return False
+
+    def is_indexing(self):
+        '''Is an indexing process currently running'''
+        if self.has_uuids() or self.errors:
+            return True
+        for client_info in self._client_conns.values():
+            if client_info['uuid_cnt']:
+                return True
```

Comment:
How about: 

return false or client_info['uuid_cnt'] or not client_info['results']

Reply:
client_info['uuid_cnt'] would return a number so we'd want, false or client_info['uuid_cnt'] > 0 or not client_info['results']

I agree this can be cleaned up.  Probably with an any() and a better boolean expression like you suggested.  I'll keep this in mind for the follow up PR given the various queue types.